### ISSUE TITLE
Line number for duplicate object keys error is incorrect

### ISSUE THREAD

**Issue by elynde** [eslint/eslint#3573]
At Facebook, our linters only warn you about issues on lines that you changed. This caused an issue for us because the line number for a duplicate key in an object had the line number of the start of the object instead of the actual duplicate key. For example:

```
var foo = { // this was where the error was reported
  bar: 1,
  // ...
  // possibly many other lines here
  // ...
  bar: 1, // this line was added
};

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by elynde** [comment]
https://github.com/eslint/eslint/pull/3569

**Comment by nzakas** [comment]
Thanks for the report. I see you've find ahead and sent a pull request, which is great. Please be sure in the future to provide all the info requested in the second current. It really helps us.

**Comment by elynde** [comment]
No problem. What's the second current?

**Comment by gyandeeps** [comment]
Second comment (which is by @eslintbot )

**Comment by elynde** [comment]
Ah ok will do!

**Comment by nzakas** [comment]
Sorry, autocorrect is pretty bad on this tablet.

**Comment by elynde** [comment]
Ah ya no problem, will do! New to Github/open-source so I'm still learning the ropes. I appreciate how friendly and responsive you folks are!
